lib/season_statistics.rb

```diff
@@ -0,0 +1,167 @@
+  end
+
+  def worst_win_percentage_by_season
+    least_number_of_games_won = @counter_wins_team_id.invert.min[0].to_f
+    for_lowest_total_games_played = @games_played_by_team_id[@counter_wins_team_id.invert.min[1]]
+    worst = (least_number_of_games_won / for_lowest_total_games_played) * 100
+  end
+
+  def winningest_and_worst_suite(season)
+    create_coach_by_team_id
+    collect_game_objects_by_season(season)
+    total_wins_by_season
+    best_win_percentage_by_season
+    worst_win_percentage_by_season
+  end
+
+  def winningest_coach(season)
+    winningest_and_worst_suite(season)
+    max_team_id = @counter_wins_team_id.invert.max[1]
+    winningest_coach = @coach_by_team_id[max_team_id]
+  end
+
+  def worst_coach(season)
+    winningest_and_worst_suite(season)
+    min_team_id = @counter_wins_team_id.invert.min[1]
+    worst_coach = @coach_by_team_id[min_team_id]
+  end
+
+  def collect_game_team_objects_by_season(season)
+    all_game_teams.each do |game_team_object|
+      if season.to_s[0..3] == game_team_object.game_id.to_s[0..3]
+        @by_season_game_team_objects << game_team_object
+      end
+    end
+  end
+
+  def shots_by_team_id_by_season
+    @shots_by_team_id = Hash.new{ |hash, key| hash[key] = 0 }
+    @by_season_game_team_objects.each do |season_game_team_object|
+      @shots_by_team_id[season_game_team_object.team_id] += season_game_team_object.shots
+    end
+  end
+
+  def goals_by_team_id_by_season
+    @goals_by_team_id = Hash.new{ |hash, key| hash[key] = 0 }
+    @by_season_game_team_objects.each do |season_game_team_object|
+      @goals_by_team_id[season_game_team_object.team_id] += season_game_team_object.goals
+    end
+  end
+
+  def shot_accuracy_by_team_id_by_season
+    @goals_by_team_id.keys.each do |key|
+      shot_accuracy = @goals_by_team_id[key].to_f / @shots_by_team_id[key]
+      @shot_accuracy_by_team_id[key] = shot_accuracy
+    end
+  end
+
+  def shot_accuracy_suite(season)
+    collect_game_team_objects_by_season(season)
+    shots_by_team_id_by_season
+    goals_by_team_id_by_season
+    shot_accuracy_by_team_id_by_season
+  end
+
+  def most_accurate_team(season)
+    shot_accuracy_suite(season)
+    @team_name_by_id[@shot_accuracy_by_team_id.invert.max[1]]
+  end
+
+  def least_accurate_team(season)
+    shot_accuracy_suite(season)
+    @team_name_by_id[@shot_accuracy_by_team_id.invert.min[1]]
+  end
+
+  def tackles_by_team_id_by_season(season)
+    collect_game_team_objects_by_season(season)
+    @by_season_game_team_objects.each do |season_game_team_object|
+      @tackles_by_team_id[season_game_team_object.team_id] += season_game_team_object.tackles
+    end
+  end
+
+  def most_tackles(season)
+    tackles_by_team_id_by_season(season)
+    @team_name_by_id[@tackles_by_team_id.invert.max[1]]
+  end
+
+  def fewest_tackles(season)
+    tackles_by_team_id_by_season(season)
+    @team_name_by_id[@tackles_by_team_id.invert.min[1]]
+  end
+end
```

Comment:
This is running a bit long. We will likely need to work on refactoring this to meet the 150 line requirement.